detection_rules/rule.py

```diff
@@ -30,14 +35,32 @@ class RuleMeta(MarshmallowDataclassMixin):
     deprecation_date: Optional[definitions.Date]
 
     # Optional fields
-    beats_version: Optional[definitions.BranchVer]
-    ecs_versions: Optional[List[definitions.BranchVer]]
```

Comment:
I wonder if it's worth keeping these around and raising a warning or a more helpful error saying those fields are ignored. Then we'll have slightly less cryptic errors if people are using those fields in a fork.